keps/sig-apps/sidecarcontainers.md

```diff
@@ -196,7 +197,14 @@ Older Kubelets that don't implement the sidecar logic could have a pod scheduled
 ## Design Details
 
 ### Test Plan
-//TODO
+* Units test in kubelet package `kuberuntime` primarily in the same style as `TestComputePodActions` to test a variety of scenarios.
+* New E2E Tests to validate that pods with sidecars behave as expected e.g:
+ * Pod with sidecars starts sidecars containers before non-sidecars
+ * Pod with sidecars terminates non-sidecar containers before sidecars
+ * Pod with init containers and sidecars starts sidecars after init phase, before non-sidecars
+ * Termination grace period is still respected when terminating a Pod with sidecars
+ * Job with sidecars completes once non-sidecars have completed
```

Comment:
What about Pod phase? If all non-sidecars terminate successfully but some sidecars terminates in a failure, Pod phase should be Succeeded or Failed? We should add a test for that. 

Also Job completion is based on # of Pods succeeded/failed. You'd probably want to test the last one against a Pod (with restartPolicy != Always) instead of a Job.

Reply:
thanks for having a look @janetkuo 
Yeah, the current implementation would have pod phase include the status of sidecars, so I've added some tests for that

Also I've changed the last test to just be against a pod with restaryPolicy != always